lib/supavisor/db_handler.ex

```diff
@@ -255,4 +264,72 @@ defmodule Supavisor.DbHandler do
 
     :keep_state_and_data
   end
+
+  # Adapted from Postgrex.Protocol
+  # TODO: Pull over handshake timeout from Postgrex.Protocol?
+  defp handshake(sock, %{ssl: true} = auth), do: ssl(sock, auth)
+  defp handshake(sock, %{ssl: false} = auth), do: send_startup(sock, auth)
```

Comment:
This mirrors a lot from Postgrex, except doesn't have some more advanced stuff that it does, like a handshake timeout